### PR TITLE
Add mobilenet.

### PR DESCRIPTION
**Pull request by nsthorat**
Adds a pre-trained MobileNet model module which lets you use a pretrained MobileNet model. It completely hides Tensors so users don't need to know anything about machine learning.

You can load mobilenet like this:
```
const model = await mobilenet.load();
```

You can classify an image element like this:
```
const classification = await model.classify(document.getElementById('img'));
```

You can get intermediate activations like this:
```
const activation = await model.infer(document.getElementById('img'), 'conv_w_1');
```

Readme link for more details:
https://github.com/tensorflow/tfjs-models/blob/2576a5132de5dc5d328ed297d37c3595b28c62ca/mobilenet/README.md

This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/4)

### CODE REVIEW

**Comment by dsmilkov** [comment]
:lgtm_strong:

---

Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[mobilenet/index.ts, line 70 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwjuQXCJQ29z-paU4o:-LBwjuQYE0nN4QgWnlb4:bu37yeg) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/index.ts#L70)):*
> ```XML
>     this.model = await tf.loadModel(this.path);
>     // Warmup the model.
>     tf.tidy(() => this.model.predict(tf.zeros([0, IMAGE_SIZE, IMAGE_SIZE, 3])));
> ```

add await before `tf.tidy()` to wait for the warmup.

---

*[mobilenet/index.ts, line 90 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkCYNBXlenW65ZmfL:-LBwkCYNBXlenW65ZmfM:b315wfs) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/index.ts#L90)):*
> ```XML
>       throw new Error(
>           `Unknown endpoint ${endpoint}. Available endpoints: ` +
>           `${this.endpoints}.`)
> ```

missing semicolon. Setup tslint with clang-format and .vscode folder that has workspace settings.

---

*[mobilenet/package.json, line 5 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkUyQARR1o8f9eqnI:-LBwkUyQARR1o8f9eqnJ:b-mhn1k7) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/package.json#L5)):*
> ```JSON
>   "version": "0.0.1",
>   "description": "Pretrained MobileNet in TensorFlow.js",
>   "main": "dist/mobilenet.js",
> ```

This should be dist/index.js. Strange that this worked for you. Did you rename mobilenet.ts to index.ts last minute?

---

*[mobilenet/package.json, line 7 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkcF24C1RBA9qX9FK:-LBwkcF24C1RBA9qX9FL:bh613yh) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/package.json#L7)):*
> ```JSON
>   "main": "dist/mobilenet.js",
>   "unpkg": "dist/bundle.js",
>   "types": "dist/mobilenet.d.ts",
> ```

dist/index.d.ts

---

*[mobilenet/README.md, line 22 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkkCqBB4EJGc5g--6:-LBwkkCqBB4EJGc5g--7:b-e8rmbx) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/README.md#L22)):*
> ```Markdown
> 
> ```html
> <html>
> ```

consider dropping the <html>, <head> and <body> tags to keep it shorter. It's implicit.

---

*[mobilenet/tsconfig.json, line 23 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwl2kZ0fUa_s48rcNl:-LBwl2kZ0fUa_s48rcNm:b-bxug8x) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/tsconfig.json#L23)):*
> ```JSON
>   },
>   "include": [
>     "mobilenet.ts", "imagenet_classes.ts"
> ```

index.ts ?

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-:-LBwl61j6jEacDGJKObb:b-jbsf8z)*

**Comment by nsthorat** [comment]
Review status: 3 of 8 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[mobilenet/index.ts, line 70 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwjuQXCJQ29z-paU4o:-LBwtceV7cUdWNJ9Cw0R:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/index.ts#L70)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

add await before `tf.tidy()` to wait for the warmup.
</blockquote></details>

Done.

---

*[mobilenet/index.ts, line 90 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkCYNBXlenW65ZmfL:-LBwtqKFB9Yz3MTNFn96:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/index.ts#L90)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

missing semicolon. Setup tslint with clang-format and .vscode folder that has workspace settings.
</blockquote></details>

Done.

---

*[mobilenet/package.json, line 5 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkUyQARR1o8f9eqnI:-LBwtr0X8l8ib71qmzRU:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/package.json#L5)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

This should be dist/index.js. Strange that this worked for you. Did you rename mobilenet.ts to index.ts last minute?
</blockquote></details>

Done.

---

*[mobilenet/package.json, line 7 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkcF24C1RBA9qX9FK:-LBwtrHW5nYblZlExcyz:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/package.json#L7)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

dist/index.d.ts
</blockquote></details>

Done.

---

*[mobilenet/README.md, line 22 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwkkCqBB4EJGc5g--6:-LBwu4z6Ajoj9vRxJ-w0:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/README.md#L22)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

consider dropping the <html>, <head> and <body> tags to keep it shorter. It's implicit.
</blockquote></details>

Done.

---

*[mobilenet/tsconfig.json, line 23 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/4#-LBwl2kZ0fUa_s48rcNl:-LBwu7lZ2RAH21Ph2___:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/1f7a6045e8af6b78502650a1bf5b205d5fe17710/mobilenet/tsconfig.json#L23)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

index.ts ?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/4)*
